common/processor/helidon-copyright/src/main/java/io/helidon/common/processor/helidon/copyright/HelidonCopyrightProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.processor.helidon.copyright;
+
+import java.time.LocalDate;
+
+import io.helidon.common.Weight;
+import io.helidon.common.processor.spi.CopyrightProvider;
+import io.helidon.common.types.TypeName;
+
+/**
+ * Java {@link java.util.ServiceLoader} provider implementation that generates copyright as used by the Helidon project.
+ */
+@Weight(100)
+public class HelidonCopyrightProvider implements CopyrightProvider {
```

Comment:
Keep it simple, and just have this implement `GeneratedAnnotationProvider` interface too.

Reply:
Why? The generated annotation is always from Helidon common. The copyright is only used for Helidon types that form interfaces. I do not think the two should be bound together